### PR TITLE
Update System.Security.Cryptography.Pkcs (dependabot missed it)

### PR DESCRIPTION
**Pull request by JanKrivanek**
Fixes CVE-2023-29331

### Context
[CVE-2023-29331](https://github.com/advisories/GHSA-555c-2p6r-68mm)

### Changes Made
Version bump that should have been done by dependabot

### CODE REVIEW

**Comment by JanKrivanek** [comment]
@MichaelSimons - Is it possible to reference `system.security.cryptography.pkcs.7.0.2` as an external nuget? This worked just fine for 7.0.0 - however I do not see 7.0.2 being supported: https://github.com/dotnet/source-build-reference-packages/tree/main/src/referencePackages/src/system.security.cryptography.pkcs Is that planned to be added? (as 7.0.0 and 7.0.1 have known vulnerabilities)

**Comment by MichaelSimons** [comment]
> @MichaelSimons - Is it possible to reference `system.security.cryptography.pkcs.7.0.2` as an external nuget? This worked just fine for 7.0.0 - however I do not see 7.0.2 being supported: https://github.com/dotnet/source-build-reference-packages/tree/main/src/referencePackages/src/system.security.cryptography.pkcs Is that planned to be added? (as 7.0.0 and 7.0.1 have known vulnerabilities)

source-build-reference-packages is a self servicing repo.  If you need a new package added which it appears like it is, you can add it yourself following these [instructions](https://github.com/dotnet/source-build-reference-packages#reference).  If you need any help I will be glad to help.

**Comment by JanKrivanek** [comment]
@dotnet/source-build-internal - Adding `System.Security.Cryptography.Pkcs/*7.0.2*` to prebuild as a temporary workaround, so that we can update the vulnerable package.

The proper solution will require SRBP update, which will require update to arcade 8 - so this will need a bit of time

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -42,5 +42,10 @@
       <Uri>https://github.com/dotnet/arcade</Uri>
       <Sha>2b9ccd49bbcca2156e229b00a140a8a018baac58</Sha>
     </Dependency>
+    <!-- Added manually - as System.Security.Cryptography.Pkcs is not darc subscribed -->
+    <Dependency Name="System.Security.Cryptography.Pkcs" Version="7.0.2">
```
Comment (rainersigwald) [inline eng/Version.Details.xml:46]: Should this be in "product" dependencies?

**Review by rainersigwald — APPROVED** [review]

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -42,5 +42,10 @@
       <Uri>https://github.com/dotnet/arcade</Uri>
       <Sha>2b9ccd49bbcca2156e229b00a140a8a018baac58</Sha>
     </Dependency>
+    <!-- Added manually - as System.Security.Cryptography.Pkcs is not darc subscribed -->
```
Comment (MichaelSimons) [inline eng/Version.Details.xml:45]: Why is this new dependency necessary?

Comment (JanKrivanek) [reply eng/Version.Details.xml]: Accidental leftover - removing

File: eng/Version.Details.xml (hunk 1 above)
Comment (JanKrivanek) [reply eng/Version.Details.xml]: Accidental leftover - removing

**Review by MichaelSimons — APPROVED** [review]

**Review by rainersigwald — APPROVED** [review]

**Pull request merged**